### ISSUE TITLE
Cidram response

### ISSUE THREAD

**Issue by mikeruss1** [CIDRAM/CIDRAM#249]
I moved to a new hosting service 4 months ago and noticed an obvious improvement in page response time, which has been slowly getting marginally worse. As an experiment I deleted cidram cache and it's back to being very snappy. 

This.is not very scientific I know, I should have monitored elapsed times. Cache.dat was about 150K. The majority being IP tracking. I have been modifying the amount of rubbish I track, but it was still substantial.

Not sure if this helps anyone, I think in future I will limit cache size to maybe 50K.

**Comment by 737simpilot** [comment]
Weren't you using PDO?

There _could_ be other factors. One of them being how the host has things set up. If I remember, you're using a host that I, as well as others, are not too keen on saying they're the best out there. I could be mistaken. Reason I say this is that I was with my first host for some four years and all of a sudden it was like CORS or their firewall or some damn thing was really messing up my ability to use CloudFlare. After days and weeks of me painstakingly trying everything I could, up to and including trying another reverse proxy, I said enough of this and moved to another host. Well, that was like a night and day difference let me tell you. The reverse proxy issue was gone, and to my surprise they were using an updated cPanel where my host wasn't. I thought my host was and I was just used to seeing that type even though there were updates and what have you, the cPanel options were more sparse and what not with my old host. Not only that, but I was always seeing odd IPs that last logged into my cPanel. I talked to my host (practically a one man outfit) and he said there was no issue. 

I couldn't stay with the new host though even thought they seemed pretty decent and I think had some really good backend Internet connections or something. The website load time was a lot faster just due to that alone. Well, the new host I had gives you a dedicated IP for shared accounts. May sound nice, but if that's the case and you're behind a reverse proxy like CloudFlare or what ever, you'll stick out like a sore thumb on Censys et al. So, I had to go and find yet another host.  Like the last new host I had, they too had the same updated cPanel and there have been no major uncorrective issues thus far unlike the old host. Kinda pisses me off that having been with the old host for so long I put up with that. When I downloaded my public directory and database I scanned the living crap out of everything with Herdprotect portable and what not. I also ran my installation in XAMPP looking for stray external IP requests that shouldn't be there with a network sniffer. All because of my old host showing odd IPs in my cPanel. I knew my host's IPs, but the others usually came back from Rogers Cable in Canada. I wish I knew what I know now about some IP investigation because I would have researched those IPs. 

To use an analogy, hosts can be as different as two Docker containers all in the same OS. 

I'm almost willing to bet that if you downloaded your site with the same cache size (now deleted) and ran it in XAMPP, you may not see that sluggishness at all. Or you could spin up a LAMP stack in VMware Workstation Player and try in there for comparison. 

Have you noticed if your your resources are being saturated at all? If it's a shared account, go [here](https://viewdns.info/) and enter your origin IP under "reverse IP." Top left. If you see an abhorrent amount of domains, your host is overselling the server resources. That's if your account is a shared account of course. To get the actual number, press and hold the control key and highlight on down the list, right click and copy. Now paste in Notepad++. Look at the line number at the end of the list. For me, my host has 302 domains for my IP which is probably within reasonable limits. My host did say that the IP or what ever I was on wasn't using a lot of resources when I had to bring an unrelated issue to them about CloudFlare potentially being blocked by their firewall. Turned out if was their Rocket Loader config. or something I can't remember now. That's actually a nice perk to have if your host is willing to pay for it as a hosting provider to CloudFlare. It may or may not increase page load times though. And static content isn't cached unless you create page rules. I don't use caching. Just the WAF/ DDoS, layer 7 DDoS mitigation ability and keeping the origin IP hidden. Probably way over the top, but A) all part and parcel of the nerd ethos, and B) anyone not liking my content can own me and I refuse to be tea bagged if I can prevent it. LOL Though, I can deploy a backup in about 30 minutes. And I backup like no one's biz. Even on DVD/RW and Blu-ray BD-RE. All in Truecrypt encrypted containers using a cascade of ciphers. :nerd_face:

**Comment by mikeruss1** [comment]
thanks Aaron, I moved from Hostgator thank goodness. Response is now excellent. Last time I tried it I couldnt get PDO to work. I have memcached now which I should really try for CIDRAM. Trouble is I would loose everything if they did a restart which would be a pain. I know its not a proper test, however deleting CIDRAM cache made an immediate and noticeable difference, so I suspect there is mileage in not letting it get too big.
